core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1168,19 +1170,15 @@ private void deleteInternal(LockedInodePath inodePath, boolean replayed, long op
     }
 
     List<Pair<AlluxioURI, Inode<?>>> delInodes = new LinkedList<>();
-    // Persisted directories which can safely be deleted recursively from the UFS
-    // Mount points are not deleted recursively as we need to preserve the directory itself
-    Map<AlluxioURI, Inode<?>> safeRecursiveUFSDeletes = new HashMap<>();
 
     Pair<AlluxioURI, Inode<?>> inodePair =
         new Pair<AlluxioURI, Inode<?>>(inodePath.getUri(), inode);
     delInodes.add(inodePair);
-    UfsSyncChecker ufsChecker = new UfsSyncChecker(mMountTable);
+    UfsSyncChecker ufsSyncChecker = new UfsSyncChecker(mMountTable, deleteOptions.isUnchecked());
```

Comment:
It seems weird to create a dummy `UfsSyncChecker` when `isUnchecked` is true. Is it possible to just skip the checks and only create the checker if `isUnchecked` is false?